public/app/features/commandPalette/KBarResults.tsx

```diff
@@ -0,0 +1,226 @@
+        // as part of the dependencies array.
+        activeRef.current?.click();
+      }
+    };
+    window.addEventListener('keydown', handler);
+    return () => window.removeEventListener('keydown', handler);
+  }, [query]);
+
+  // destructuring here to prevent linter warning to pass
+  // entire rowVirtualizer in the dependencies array.
+  const { scrollToIndex } = rowVirtualizer;
+  React.useEffect(() => {
+    scrollToIndex(activeIndex, {
+      // ensure that if the first item in the list is a group
+      // name and we are focused on the second item, to not
+      // scroll past that group, hiding it.
+      align: activeIndex <= 1 ? 'end' : 'auto',
+    });
+  }, [activeIndex, scrollToIndex]);
+
+  React.useEffect(() => {
+    // TODO(tim): fix scenario where async actions load in
+    // and active index is reset to the first item. i.e. when
+    // users register actions and bust the `useRegisterActions`
+    // cache, we won't want to reset their active index as they
+    // are navigating the list.
+    query.setActiveIndex(
+      // avoid setting active index on a group
+      typeof props.items[START_INDEX] === 'string' ? START_INDEX + 1 : START_INDEX
+    );
+  }, [search, currentRootActionId, props.items, query]);
+
+  const execute = React.useCallback(
+    (ev: React.MouseEvent, item: RenderParams['item']) => {
+      if (typeof item === 'string') {
+        return;
+      }
+
+      // ActionImpl constructor copies all properties from action onto ActionImpl
+      // so our url property is secretly there, but completely untyped
+      // Preferably this change is upstreamed and ActionImpl has this
+      // eslint-disable-next-line
+      const url = (item as ActionImpl & { url?: string }).url;
+
+      if (item.command) {
+        item.command.perform(item);
+        query.toggle();
+      } else if (url) {
+        if (!(ev.ctrlKey || ev.metaKey || ev.shiftKey)) {
+          query.toggle();
+        }
+      } else {
+        query.setSearch('');
+        query.setCurrentRootAction(item.id);
+      }
+
+      options.callbacks?.onSelectAction?.(item);
+    },
+    [query, options]
+  );
+
+  const pointerMoved = usePointerMovedSinceMount();
+
+  return (
+    <div
+      ref={parentRef}
+      style={{
+        maxHeight: props.maxHeight || 400,
+        position: 'relative',
+        overflow: 'auto',
+      }}
+    >
+      <div
+        role="listbox"
+        id={KBAR_LISTBOX}
+        style={{
+          height: `${rowVirtualizer.totalSize}px`,
+          width: '100%',
+        }}
+      >
+        {rowVirtualizer.virtualItems.map((virtualRow) => {
+          const item = itemsRef.current[virtualRow.index];
+
+          // ActionImpl constructor copies all properties from action onto ActionImpl
+          // so our url property is secretly there, but completely untyped
+          // Preferably this change is upstreamed and ActionImpl has this
+          // eslint-disable-next-line
```

Comment:
think we should be explicit about the rule we're ignoring here
```suggestion
          // eslint-disable-next-line @typescript-eslint/consistent-type-assertions
```